internal/v4/search.go

```diff
@@ -137,17 +135,24 @@ func parseSearchParams(req *http.Request) (charmstore.SearchParams, error) {
 	return sp, nil
 }
 
+// groupsFromRequest finds the Read ACL users and groups that should be added to the
+// SearchParams.Groups list. It will also determine if the requestee has full admin rights.
 func (h *Handler) groupsFromRequest(sp *charmstore.SearchParams, req *http.Request) {
```

Comment:
I kind of feel that this doesn't need to be tied so closely to SearchParams.
It could be groupsFromRequest(req *http.Request) (isAdmin bool, groups []string, err error)
and which is closer to the contract implied by the name.

But when you look at it that way, it's not really much different from checkRequest,
which takes a request and returns some user info.

So perhaps better still might be to change checkRequest to check for admin credentials,
and add Admin to authDeclaredAttrs (authInfo might be a better name for that).

Then checkBasicAuth only needs to be called in one place.